### PR TITLE
Feature/e UI 4858 page flow

### PR DESCRIPTION
**Pull request by rayliangatos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4858

### Change description ###
Build Page flow engine for request hearings, depends on setting coming from service API.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  let fixture: ComponentFixture<HearingCheckAnswersComponent>;
+  const mockedHttpClient = jasmine.createSpyObj('HttpClient', ['get', 'post']);
+  const hearingsService = new HearingsService(mockedHttpClient);
+  hearingsService.navigateAction$ = of(ACTION.CONTINUE);
+
+  const initialState = {
+    hearings: {
+      hearingList: {
+        caseHearingMainModel: [
+          {
```
Comment (waynepHMCTS) [inline src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.spec.ts:19]: indentation

File: src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  templateUrl: './hearing-check-answers.component.html',
+})
+export class HearingCheckAnswersComponent extends RequestHearingPageFlow implements OnInit, OnDestroy {
+
+  constructor(protected readonly hearingStore: Store<fromHearingStore.State>,
+              protected readonly hearingsService: HearingsService) {
+    super(hearingStore, hearingsService);
+  }
+
+  public ngOnInit(): void {
```
Comment (waynepHMCTS) [inline src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.ts:19]: not needed

File: src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+})
+export class HearingFacilitiesComponent extends RequestHearingPageFlow implements OnInit, OnDestroy {
+
+  constructor(protected readonly hearingStore: Store<fromHearingStore.State>,
+              protected readonly hearingsService: HearingsService) {
+    super(hearingStore, hearingsService);
+  }
+
+  public ngOnInit(): void {
+  }
```
Comment (waynepHMCTS) [inline src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts:20]: not needed

File: src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  public executeAction(action: ACTION): void {
+    if (this.isFormValid()) {
+      super.navigateAction(action);
+    }
+  }
+
+  public isFormValid(): boolean {
+    // TODO verify if form group is valid
```
Comment (waynepHMCTS) [inline src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts:29]: unnecessary comment

File: src/hearings/containers/request-hearing/hearing-additional-instructions/hearing-additional-instructions.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  let fixture: ComponentFixture<HearingAdditionalInstructionsComponent>;
+  const mockedHttpClient = jasmine.createSpyObj('HttpClient', ['get', 'post']);
+  const hearingsService = new HearingsService(mockedHttpClient);
+  hearingsService.navigateAction$ = of(ACTION.CONTINUE);
+
+  const initialState = {
+    hearings: {
+      hearingList: {
+        caseHearingMainModel: [
+          {
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-additional-instructions/hearing-additional-instructions.component.spec.ts:19]: can't see any indention issue here, could you double check.

File: src/hearings/containers/request-hearing/hearing-attendance/hearing-attendance.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  let fixture: ComponentFixture<HearingAttendanceComponent>;
+  const mockedHttpClient = jasmine.createSpyObj('HttpClient', ['get', 'post']);
+  const hearingsService = new HearingsService(mockedHttpClient);
+  hearingsService.navigateAction$ = of(ACTION.CONTINUE);
+
+  const initialState = {
+    hearings: {
+      hearingList: {
+        caseHearingMainModel: [
+          {
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-attendance/hearing-attendance.component.spec.ts:19]: can't see any indention issue here, could you double check.

File: src/hearings/containers/request-hearing/hearing-attendance/hearing-attendance.component.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  public executeAction(action: ACTION): void {
+    if (this.isFormValid()) {
+      super.navigateAction(action);
+    }
+  }
+
+  public isFormValid(): boolean {
+    // TODO verify if form group is valid
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-attendance/hearing-attendance.component.ts:29]: I deliberately add this // TODO comment, as when implement the component it needs to follow the same pattern for form validation under the page flow. This // TODO should be removed after the validation is added but not now.

File: src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.spec.ts (hunk 1 above)
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.spec.ts:19]: can't see any indention issue here, could you double check.

File: src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts (hunk 4 above)
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts:29]: I deliberately add this // TODO comment, as when implement the component it needs to follow the same pattern for form validation under the page flow. This // TODO should be removed after the validation is added but not now.

File: src/hearings/containers/request-hearing/hearing-judge/hearing-judge.component.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  public executeAction(action: ACTION): void {
+    if (this.isFormValid()) {
+      super.navigateAction(action);
+    }
+  }
+
+  public isFormValid(): boolean {
+    // TODO verify if form group is valid
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-judge/hearing-judge.component.ts:29]: I deliberately add this // TODO comment, as when implement the component it needs to follow the same pattern for form validation under the page flow. This // TODO should be removed after the validation is added but not now.

File: src/hearings/containers/request-hearing/hearing-panel/hearing-panel.component.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  public executeAction(action: ACTION): void {
+    if (this.isFormValid()) {
+      super.navigateAction(action);
+    }
+  }
+
+  public isFormValid(): boolean {
+    // TODO verify if form group is valid
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-panel/hearing-panel.component.ts:29]: I deliberately add this // TODO comment, as when implement the component it needs to follow the same pattern for form validation under the page flow. This // TODO should be removed after the validation is added but not now.

File: src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(protected readonly hearingStore: Store<fromHearingStore.State>,
+              protected readonly hearingsService: HearingsService) {
+    super(hearingStore, hearingsService);
+  }
+
+  public ngOnInit(): void {
+  }
+
+  public executeAction(action: ACTION): void {
+    if (this.isFormValid()) {
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts:23]: Yes this is finally needed for form group validation, please check the hearing-timing for how this is used.

File: src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.ts (hunk 2 above)
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-check-answers/hearing-check-answers.component.ts:19]: I left this as this component is not finally built. But we will remove it if we don't use the ngOnInit method finally at later point.

File: src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts (hunk 3 above)
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-facilities/hearing-facilities.component.ts:20]: I left this as this component is not finally built. But we will remove it if we don't use the ngOnInit method finally at later point.

File: src/hearings/containers/request-hearing/hearing-requirements/hearing-requirements.component.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  templateUrl: './hearing-requirements.component.html',
+})
+export class HearingRequirementsComponent extends RequestHearingPageFlow implements OnInit, OnDestroy {
+
+  constructor(protected readonly hearingStore: Store<fromHearingStore.State>,
+              protected readonly hearingsService: HearingsService) {
+    super(hearingStore, hearingsService);
+  }
+
+  public ngOnInit(): void {
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/hearing-requirements/hearing-requirements.component.ts:19]: I left this as this component is not finally built. But we will remove it if we don't use the ngOnInit method finally at later point.

**Review by waynepHMCTS — APPROVED** [review]

**Review by Kartheek-S — APPROVED** [review]

**Pull request merged**